airflow/models/__init__.py

```diff
@@ -4872,11 +4872,12 @@ def get_previous_scheduled_dagrun(self, session=None):
         ).first()
 
     @provide_session
-    def update_state(self, session=None):
+    def update_state(self, session=None, finished_tasks=None):
         """
         Determines the overall state of the DagRun based on the state
         of its TaskInstances.
 
+        :param finished_tasks: The finished tasks collected ordered by dagrun as a column (task_name, state)
```

Comment:
The second thing this function does (line 4885) is get all the task instances for the run, so I don't see the need to pass finished_tasks here. Passing it on to the DepContext makes sense though.

Reply:
It is necessary for the dependency check of all unfinished tasks(4915), and we have it already so why not.